### PR TITLE
Documentation for AWS DynamoDB guide

### PR DESCRIPTION
**Pull request by greedy52**
Related PR
- #10025
- #14429

### CODE REVIEW

**Comment by ptgott** [comment]
Hi @greedy52, would you mind separating the DynamoDB guide into its own PR? This will make your changes much easier/faster to review. Thanks!

**Comment by greedy52** [comment]
> Hi @greedy52, would you mind separating the DynamoDB guide into its own PR? This will make your changes much easier/faster to review. Thanks!

@ptgott done! Other PR #14429. Thanks!

File: docs/config.json [hunk 1]
```diff
@@ -506,6 +506,10 @@
               "title": "AWS ElastiCache & MemoryDB",
               "slug": "/database-access/guides/redis-aws/"
             },
+            {
+              "title": "AWS DynamoDB",
```
Comment (ptgott) [inline docs/config.json:510]: This seems like an Application Service guide—should it go in that section instead?

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 2]
```diff
@@ -0,0 +1,298 @@
+---
+title: Database Access with AWS DynamoDB
+description: How to access AWS DynamoDB with Teleport.
+---
+
+<Admonition type="note" title="Important">
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:6]: I think we should move this guide to the Application Access section and remove this admonition. This way, we can keep all guides related to a single Teleport resource service in the same place for better discoverability (and also to illustrate all of the capabilities of Application Access).

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+Access to AWS DynamoDB is provided by [**Teleport Application Access**](../../application-access/introduction.mdx) for AWS Console and API.
+</Admonition>
+
+This guide will help you to:
+- Install Teleport Application Service.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:11]:
```suggestion
- Install the Teleport Application Service.
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+- Set up Teleport to access AWS Console and API.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:12]:
```suggestion
- Set up Teleport to access the AWS Console and API.
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+- Connect to your DynamoDB through Teleport.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:13]:
```suggestion
- Connect to your DynamoDB databases through Teleport.
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+![DynamoDB Self-Hosted](../../../img/database-access/guides/dynamodb_selfhosted.png)
+</ScopedBlock>
+
+<ScopedBlock scope={["cloud"]}>
+![DynamoDB Cloud](../../../img/database-access/guides/dynamodb_cloud.png)
+</ScopedBlock>
+
+## Prerequisites
+
+- A running Teleport cluster, either self hosted or in Teleport Cloud.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:25]: We can remove this since this includes the `edition-prereqs-tabs.mdx` partial below.

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 7]
```diff
@@ -0,0 +1,298 @@
...
+
+We will assume your Teleport cluster is accessible at `teleport.example.com` and `*.teleport.example.com`. You can substitute the address of your Teleport Proxy Service. (For Teleport Cloud customers, this will be similar to `mytenant.teleport.sh`.)
+
+<Admonition type="note" title="Application Access and DNS">
+(!docs/pages/includes/dns-app-access.mdx!)
+</Admonition>
+
+## Step 1/5. Create an IAM role for DynamoDB access
+
+The first step is to create an IAM role with desired IAM permissions that you wish to grant to a Teleport user. In this example, you will create an IAM role with full DynamoDB access. **Skip this step** if you already have the roles you want to provide access to.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:47]: Which IAM permissions are required for a Teleport user here? Are there any guidelines we can offer?

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 8]
```diff
@@ -0,0 +1,298 @@
...
+
+Now go to the IAM -> Access Management ->
+[Roles](https://console.aws.amazon.com/iamv2/home#/roles), then press Create Role.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:50]: I think we can replace this entire flow with this link, e.g., "Now visit the [Roles page] of the AWS Console..."

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 9]
```diff
@@ -0,0 +1,298 @@
...
+
+Select the "AWS account" option which creates a default [trust
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:52]:
```suggestion
Select the "AWS account" option, which creates a default [trust
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+policy](https://aws.amazon.com/blogs/security/how-to-use-trust-policies-with-iam-roles/)
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:53]: I think we can remove this link to avoid breaking the reader's flow

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+to allow other entities in this account to assume this role.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:54]:
```suggestion
to allow other entities in this account to assume this role:
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+
+![Create Role Step 1](../../../img/database-access/guides/dynamodb-create-role-1.png)
+
+Press Next. Find the AWS managed policy `AmazonDynamoDBFullAccess` by Filter and then select the policy.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:58]:
```suggestion
Press Next. Find the AWS-managed policy `AmazonDynamoDBFullAccess` and then select the policy.
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+
+![Create Role Step 2](../../../img/database-access/guides/dynamodb-create-role-2.png)
+
+Press Next. Enter a role name and press Create role.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:62]: I think it would be a bit clearer to use either bold formatting or double quotes to separate the link text the reader should click on from the rest of the text.

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+
+![Create Role Step 3](../../../img/database-access/guides/dynamodb-create-role-3.png)
+
+## Step 2/5. Configure Teleport IAM role mapping
+The next step is to give your Teleport users permissions to assume IAM roles in
+your Teleport cluster.
+
+You can do this by creating a role with the `aws_role_arns` field listing all IAM
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:70]: I found this sentence a bit hard to parse. Is there another way to phrase it?

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+  name: aws-dynamodb-access
+spec:
+  allow:
+    app_labels:
+      '*': '*'
+    aws_role_arns:
+    - arn:aws:iam::123456789000:role/ExampleTeleportDynamoDBRole
+```
+
+The `aws_role_arns` field supports template variables so they can be populated
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:86]: If this paragraph isn't necessary for following the instructions in this tutorial, I think we can either remove it, move the link to the "Next steps" section, or add it to a `Details` box.

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 16]
```diff
@@ -0,0 +1,298 @@
...
+dynamically based on your users' identity provider attributes. See [Role
+Templates](../../access-controls/guides/role-templates.mdx) for details.
+
+Now assign this role to the Teleport users you wish to grant the AWS
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:90]: I think we should rephrase this to a sentence that is more directly related to this guide:

```markdown
Now assign this role to the Teleport users you wish to
grant access to DynamoDB.
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 17]
```diff
@@ -0,0 +1,298 @@
...
+application access.
+
+## Step 3/5. Install Teleport Application Service
+
+### Generate a token
+
+A join token is required to authorize a Teleport Application Service agent to
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:97]:
```suggestion
A join token is required to authorize a Teleport Application Service instance to
```
Since some prospective users balk at "agent"

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 18]
```diff
@@ -0,0 +1,298 @@
...
+$ tctl tokens add \
+    --type=app \
+    --app-name=aws-dynamodb \
+    --app-uri=https://console.aws.amazon.com/dynamodbv2/home
+```
+
+The output should contain a `teleport app start` command that can be used to
+start the Teleport Application Service in the next step.
+
+### Install and Start Teleport
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:111]:
```suggestion
### Install and start Teleport
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 19]
```diff
@@ -0,0 +1,298 @@
...
+   --ca-pin=<ca-pin> \
+   --auth-server=https://teleport.example.com:443 \
+   --name=aws-dynamodb \
+   --uri=https://console.aws.amazon.com/dynamodbv2/home
+```
+
+## Step 4/5. Give Teleport permissions to assume roles
+
+Next, attach the following policy to the IAM role or IAM user the Teleport
+Application Service agent is using, which allows the Teleport agent to assume
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:132]:
```suggestion
Application Service instance is using, which allows the Application Service to assume
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 20]
```diff
@@ -0,0 +1,298 @@
...
+      "Action": "sts:AssumeRole",
+      "Resource": "*"
+    }
+  ]
+}
+```
+
+<Admonition type="tip">
+You can make the policy more strict by providing specific IAM role resource
+ARNs in the Resource field instead of using a wildcard.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:150]:
```suggestion
ARNs in the "Resource" field instead of using a wildcard.
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 21]
```diff
@@ -0,0 +1,298 @@
...
+</Admonition>
+
+## Step 5/5. Connect
+
+Once the Application Service has started and joined the cluster, you can start
+connecting to your DynamoDB database.
+
+### Using AWS Management Console
+
+First log in to the Teleport Web Interface at https://teleport.example.com (replace
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:160]:
```suggestion
First log in to the Teleport Web UI at `https://teleport.example.com` (replace
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 22]
```diff
@@ -0,0 +1,298 @@
...
+with your Proxy Service's public address).
+
+Navigate to the Applications tab in your Teleport cluster's control panel and
+click on the Launch button for the AWS DynamoDB application which will bring up
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:164]:
```suggestion
click on the Launch button for the AWS DynamoDB application. This will bring up
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 23]
```diff
@@ -0,0 +1,298 @@
...
+an IAM role selector:
+
+![IAM role selector](../../../img/database-access/guides/dynamodb-select-iam-role.png)
+
+Click on the role you want to assume and you will get redirected to the AWS
+management console, signed in with the selected role.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:170]:
```suggestion
Management Console, signed in with the selected role.
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 24]
```diff
@@ -0,0 +1,298 @@
...
+
+In the console's top-right corner you should see that you're logged in through
+federated login and the name of your assumed IAM role:
+
+![Federated login](../../../img/database-access/guides/dynamodb-federated-login.png)
+
+Note that your federated login session is marked with your Teleport username.
+
+### Using AWS CLI
+Before beginning this step, make sure that the `aws` command line interface (CLI) tool is installed in PATH. For more information, read [Installing or updating the latest version of the AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html).
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:180]: Should this be in the Prerequisites section?

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 25]
```diff
@@ -0,0 +1,298 @@
...
+
+```code
+$ tsh app login --aws-role ExampleTeleportDynamoDBRole aws-dynamodb
+Logged into AWS app aws. Example AWS CLI command:
+
+$ tsh aws s3 ls
+```
+
+The `--aws-role` flag allows you to specify the AWS IAM role to assume when
+accessing AWS API. You can either provide a role name like `--aws-role
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:192]:
```suggestion
accessing the AWS API. You can either provide a role name like `--aws-role
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 26]
```diff
@@ -0,0 +1,298 @@
...
+ExampleTeleportDynamoDBRole` or a full role ARN
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:193]:
```suggestion
ExampleTeleportDynamoDBRole` or a full role ARN like
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 27]
```diff
@@ -0,0 +1,298 @@
...
+`arn:aws:iam::123456789000:role/ExampleTeleportDynamoDBRole`
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:194]:
```suggestion
`arn:aws:iam::123456789000:role/ExampleTeleportDynamoDBRole`.
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 28]
```diff
@@ -0,0 +1,298 @@
...
+
+Now you can use the `tsh aws` command like the native `aws` command-line tool:
+
+```code
+$ tsh aws dynamodb list-tables
+```
+
+To log out of the aws application and remove credentials:
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:202]:
```suggestion
To log out of the `aws` application and remove credentials:
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 29]
```diff
@@ -0,0 +1,298 @@
...
+
+```code
+$ tsh app logout aws-dynamodb
+```
+
+### Using other DynamoDB applications
+
+First, log into the previously configured AWS DynamoDB app if you haven't
+already done so.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:211]:
```suggestion
already done so:
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 30]
```diff
@@ -0,0 +1,298 @@
...
+$ tsh app login --aws-role ExampleTeleportDynamoDBRole aws-dynamodb
+```
+
+To connect your DynamoDB application, you can start either a local HTTPS proxy
+or a local AWS Service Endpoint proxy.
+
+<Tabs>
+  <TabItem label="HTTPS proxy">
+  By default, starting the AWS app proxy creates a local HTTPS proxy server
+  that forwards received AWS requests to Teleport backend for AWS application
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:223]:
```suggestion
  that forwards AWS requests to the Teleport Proxy Service, enabling you to access AWS applications.
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 31]
```diff
@@ -0,0 +1,298 @@
...
+  access.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:224]:
```suggestion
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 32]
```diff
@@ -0,0 +1,298 @@
...
+    AWS_ACCESS_KEY_ID=8cd7ce58b871425e2e05
+    AWS_SECRET_ACCESS_KEY=0e08deb6ceb7538754567286922e7ada75021c40
+    AWS_CA_BUNDLE=/Users/sampleuser/.tsh/keys/teleport.example.com/alice-app/teleport.example.com/aws-dynamodb-localca.pem
+    HTTPS_PROXY=http://127.0.0.1:23456
+  ```
+
+  Use the displayed AWS credentials and HTTPS proxy settings when configuring
+  your application.
+
+  For example, you can configure the AWS credentials and the HTTPS proxy as
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:243]:
```suggestion
  For example, you can assign the AWS credentials and the HTTPS proxy address to
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 33]
```diff
@@ -0,0 +1,298 @@
...
+  environment variables for AWS SDK for Python:
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:244]:
```suggestion
  environment variables for the Python AWS SDK:
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 34]
```diff
@@ -0,0 +1,298 @@
...
+  $ python3
+  >>> import boto3
+  >>> boto3.client('dynamodb').list_tables()
+  {'TableNames': ['my-dynamodb-table'], 'ResponseMetadata': {...}}
+
+  ```
+
+  </TabItem>
+  <TabItem label="AWS Service Endpoint proxy">
+  If your application cannot use a HTTPS proxy, start the AWS app proxy with
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:259]:
```suggestion
  If your application cannot use an HTTPS proxy, start the AWS app proxy with
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 35]
```diff
@@ -0,0 +1,298 @@
...
+  the `--endpoint-url` flag to create a local server that can be used as an [AWS
+  Service Endpoint](https://docs.aws.amazon.com/general/latest/gr/rande.html):
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:261]: I think we should move this link to the "Next steps" section to avoid breaking the reader's flow

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 36]
```diff
@@ -0,0 +1,298 @@
...
+  Started AWS proxy which serves as an AWS endpoint URL at https://localhost:23457
+
+  In addition to the endpoint URL, use the following credentials to connect to the proxy:
+    AWS_ACCESS_KEY_ID=8cd7ce58b871425e2e05
+    AWS_SECRET_ACCESS_KEY=0e08deb6ceb7538754567286922e7ada75021c40
+    AWS_CA_BUNDLE=/Users/sampleuser/.tsh/keys/teleport.example.com/alice-app/teleport.example.com/aws-dynamodb-localca.pem
+  ```
+
+  For example, to connect the GUI tool
+  [dynamodb-admin](https://github.com/aaronshaf/dynamodb-admin) to the local
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:274]: Same with this link

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 37]
```diff
@@ -0,0 +1,298 @@
...
+  database endpoint:     https://127.0.0.1:23457
+  region:                ca-central-1
+  accessKey:             8cd7ce58b871425e2e05
+
+  dynamodb-admin listening on http://localhost:8001 (alternatively http://0.0.0.0:8001)
+  ```
+
+  </TabItem>
+</Tabs>
+To log out of the aws application and remove credentials:
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:291]:
```suggestion
To log out of the `aws-dynamodb` application and remove credentials:
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 38]
```diff
@@ -0,0 +1,298 @@
...
+
+```code
+$ tsh app logout aws-dynamodb
+```
+
+## Next steps
+- Take a closer look at [AWS Management and API with Teleport Application Access](../../application-access/guides/aws-console.mdx).
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:298]:
```suggestion
- More information on [AWS Management and API with Teleport Application Access](../../application-access/guides/aws-console.mdx).
```

File: docs/pages/database-access/guides/dynamodb.mdx (hunk 2 above)
Comment (greedy52) [reply docs/pages/database-access/guides/dynamodb.mdx:6]: Totally agree this guide is not very coherent with other database guides and it feels like it belongs to Application Access. I believe the original reason to have this guide is customers didn't know Teleport supports `DynamoDB`. If they have been reading the application guide on AWS, they probably can figure this out already. I feel it's more natural for ppl look for a `DynamoDB` guide in `Database Access` section.

I am up for any suggestions to finalize this guide.

@ptgott @r0mant

Comment (r0mant) [reply docs/pages/database-access/guides/dynamodb.mdx:6]: Yeah I think it belongs among the database access guides.

File: docs/pages/database-access/guides/dynamodb.mdx (hunk 7 above)
Comment (greedy52) [reply docs/pages/database-access/guides/dynamodb.mdx]: Added one sentence that mentions some AWS-managed policies and copied a note from the original guide. Not sure if this is clearer. Or let me any suggestions. Thanks.

File: docs/pages/database-access/guides/dynamodb.mdx (hunk 14 above)
Comment (greedy52) [reply docs/pages/database-access/guides/dynamodb.mdx]: I removed the last part as I think the information is repeated.

File: docs/pages/database-access/guides/dynamodb.mdx (hunk 36 above)
Comment (greedy52) [reply docs/pages/database-access/guides/dynamodb.mdx]: I removed this link. It's a third-party tool so I think it's weird to have it in "Next Steps" section. The link is the first result google returns when searching `dynamodb-admin` so should be okay not to have the link.

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 39]
```diff
@@ -0,0 +1,308 @@
...
+  connecting to:
+
+  ```code
+  $ tsh proxy aws -p 23456
+  Started AWS proxy on http://127.0.0.1:23456.
+
+  Use the following credentials and HTTPS proxy setting to connect to the proxy:
+    AWS_ACCESS_KEY_ID=<access-key-id>
+    AWS_SECRET_ACCESS_KEY=<secret-access-key>
+    AWS_CA_BUNDLE=<ca-bundle-path>
```
Comment (greedy52) [inline docs/pages/database-access/guides/dynamodb.mdx:246]: Suggested from other PR to use a placeholder instead of some random hash. Is this way like `<xxx-xx>` ok for the placeholder? or i can do the fake version like `aaaaa-bbbb-cccc"

File: docs/pages/database-access/guides/dynamodb.mdx (hunk 2 above)
Comment (ptgott) [reply docs/pages/database-access/guides/dynamodb.mdx:6]: That reasoning makes sense!

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 40]
```diff
@@ -0,0 +1,308 @@
+---
+title: Database Access with AWS DynamoDB
+description: How to access AWS DynamoDB with Teleport.
+---
+
+<Admonition type="note" title="Important">
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:6]: I think it's okay to remove this Admonition. We can work this information into the introductory content.

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 41]
```diff
@@ -0,0 +1,308 @@
...
+Access to AWS DynamoDB is provided by [**Teleport Application Access**](../../application-access/introduction.mdx) for AWS Console and API.
+</Admonition>
+
+This guide will help you to:
+- Install the Teleport Application Service.
+- Set up Teleport to access the AWS Console and API.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:12]:
```suggestion
- Set up the Teleport Application Service to access the AWS Console and API.
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 42]
```diff
@@ -0,0 +1,308 @@
...
+- Connect to your DynamoDB databases through Teleport.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:13]:
```suggestion
- Connect to your DynamoDB databases through the Teleport Application Service.
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 43]
```diff
@@ -0,0 +1,308 @@
...
+
+(!docs/pages/includes/tctl.mdx!)
+
+<Admonition type="tip" title="Not yet a Teleport user?">
+If you have not yet deployed the Auth Service and Proxy Service, you should follow one of our [getting started guides](../getting-started.mdx) or try our Teleport Application Access [interactive learning track](https://play.instruqt.com/teleport/invite/rgvuva4gzkon).
+</Admonition>
+
+We will assume your Teleport cluster is accessible at `teleport.example.com` and `*.teleport.example.com`. You can substitute the address of your Teleport Proxy Service. (For Teleport Cloud customers, this will be similar to `mytenant.teleport.sh`.)
+
+<Admonition type="note" title="Application Access and DNS">
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:42]:
```suggestion
<Admonition type="note" title="Application Access and DNS" scope={["oss", "enterprise"]} scopeOnly>
```
Suggested hiding this for Cloud users so they don't create unnecessary DNS records.

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 44]
```diff
@@ -0,0 +1,308 @@
...
+(!docs/pages/includes/dns-app-access.mdx!)
+</Admonition>
+
+## Step 1/5. Create an IAM role for DynamoDB access
+
+The first step is to create an IAM role with desired IAM permissions that you
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:48]: I still think this paragraph isn't specific enough as to which permissions a user should grant. Are there any minimum permissions we can list? 

I would suggest rephrasing this step to tell the reader to define a role with the minimum required permissions, list what these are, then include a `Details` box (or some other aside) indicating that a reader can add more permissions to the role. We should also mention some situations in which the reader would want to define a more permissive role.

**An alternative** would be to remove Step 1 and add an item to the Prerequisites list for an IAM role for DynamoDB access. Since the current instructions are telling the reader to create any role with DynamoDB permissions, I think those instructions basically amount to a prerequisite.

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 45]
```diff
@@ -0,0 +1,308 @@
...
+Press "Next". Enter a role name and press "Create role":
+
+![Create Role Step 3](../../../img/database-access/guides/dynamodb-create-role-3.png)
+
+## Step 2/5. Configure Teleport IAM role mapping
+The next step is to give your Teleport users permissions to assume IAM roles in
+your Teleport cluster.
+
+You can do this by creating a Teleport role with the `aws_role_arns` field
+listing the IAM role ARN created in the previous step.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:81]:
```suggestion
listing the IAM role ARN created in the previous step:
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 46]
```diff
@@ -0,0 +1,308 @@
...
+  name: aws-dynamodb-access
+spec:
+  allow:
+    app_labels:
+      '*': '*'
+    aws_role_arns:
+    - arn:aws:iam::123456789000:role/ExampleTeleportDynamoDBRole
+```
+
+<Details title="Templating `aws_role_arns`" opened={false}>
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:96]:
```suggestion
<Details title="Templating aws_role_arns" opened={false}>
```
Backticks aren't rendered in Details box titles

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 47]
```diff
@@ -0,0 +1,308 @@
...
+The `aws_role_arns` field supports template variables so they can be populated
+dynamically based on your users' identity provider attributes. See [Role
+Templates](../../access-controls/guides/role-templates.mdx) for details.
+</Details>
+
+Now assign this role to the Teleport users you wish to grant access to
+DynamoDB.
+
+## Step 3/5. Install Teleport Application Service
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:105]:
```suggestion
## Step 3/5. Install the Teleport Application Service
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 48]
```diff
@@ -0,0 +1,308 @@
...
+start the Teleport Application Service in the next step.
+
+### Install and start Teleport
+Install Teleport on the host where you will run the Teleport Application
+Service. See our [Installation](../../installation.mdx) page for options
+besides Linux servers.
+
+(!docs/pages/includes/install-linux.mdx!)
+
+Now start Teleport Application Service using the output from the previous step:
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:130]:
```suggestion
Now start the Teleport Application Service using the output from the previous step:
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 49]
```diff
@@ -0,0 +1,308 @@
...
+
+```code
+$ teleport app start \
+   --token=<token> \
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:134]:
```suggestion
   --token=(=presets.tokens.first=) \
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 50]
```diff
@@ -0,0 +1,308 @@
...
+   --ca-pin=<ca-pin> \
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:135]:
```suggestion
   --ca-pin=(=presets.ca_pin=) \
```

File: docs/pages/database-access/guides/dynamodb.mdx (hunk 44 above)
Comment (greedy52) [reply docs/pages/database-access/guides/dynamodb.mdx:47]: @ptgott. I believe what minimum permissions to include is very subjective to the use case. For example, if a DevOp only needs to see what tables are there, then it could be a single `ListTables`. Or if a developer is working on a specific table, the policy can restrict like `GetItem/PutItem` to that specific resource. So it's hard to come up with a list of "common" minimum required permissions.

For readers who are very experienced with AWS IAM, they should be able to refine the permissions without extra instructions here.

For those readers that are not familiar with IAM, these pre-defined AWS-mangaed policies are the easiest.

I rephrased a little bit but not sure if makes a little better. I am up for any suggestions.

Thanks!

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 51]
```diff
@@ -0,0 +1,311 @@
+---
+title: Database Access with AWS DynamoDB
+description: How to access AWS DynamoDB with Teleport.
+---
+
+Access to AWS DynamoDB is provided by [**Teleport Application
+Access**](../../application-access/introduction.mdx) for AWS Console and API.
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:7]:
```suggestion
Access**](../../application-access/introduction.mdx) for the AWS Console and API.
```

File: docs/pages/database-access/guides/dynamodb.mdx [hunk 52]
```diff
@@ -0,0 +1,311 @@
...
+
+<ScopedBlock scope={["cloud"]}>
+![DynamoDB Cloud](../../../img/database-access/guides/dynamodb_cloud.png)
+</ScopedBlock>
+
+## Prerequisites
+
+- AWS account with DynamoDB databases.
+- IAM permissions to create IAM roles.
+- `aws` (AWS Command Line Interface) installed and added to your system's
```
Comment (ptgott) [inline docs/pages/database-access/guides/dynamodb.mdx:26]: Should "system's" be "shell's"?

File: docs/pages/database-access/guides/dynamodb.mdx (hunk 44 above)
Comment (ptgott) [reply docs/pages/database-access/guides/dynamodb.mdx:47]: One issue with this approach is that, for a user who's not familiar with IAM, the instructions are vague. The only advice for setting up RBAC is:

> For access to DynamoDB, AWS provides IAM polices like AmazonDynamoDBReadOnlyAccess and AmazonDynamoDBFullAccess that can be incorporated in to roles. This example will create an IAM role with the AmazonDynamoDBFullAccess policy. Nonetheless, you can also use other AWS-managed polices or your own custom polices depending on your circumstances.

A beginner to IAM would need to know which specific policies to add, which requires finding an IAM policy reference on their own. If the user needs to find additional docs anyway, I would rather list a few good links in the Prerequisites section than include incomplete instructions in one of the steps.

What would you think of this alternative?
- Removing the first three paragraphs of this section and preserving the instructions to use the predefined `AmazonDynamoDBFullAccess` policy
- If this policy is too permissive for production, saying so in an Admonition at the end of this section and linking to an IAM policy reference

**Review by ptgott — COMMENTED** [review]
I think once we work out the question of how to address IAM policies here, this should be good to go.

File: docs/pages/database-access/guides/dynamodb.mdx (hunk 44 above)
Comment (greedy52) [reply docs/pages/database-access/guides/dynamodb.mdx]: I think this way greatly simplifies the step. I have updated this section please take another look. Thanks so much for working through this!

File: docs/pages/database-access/guides/dynamodb.mdx (hunk 52 above)
Comment (greedy52) [reply docs/pages/database-access/guides/dynamodb.mdx]: This is a very tricky one. Most of DB guides use `your system's PATH`. I think you are right most people updates `PATH` in their profile/shell. Though technically you can also update system's PATH. And, for windows, system's PATH is the only option.

I updated the line to
```
- `aws` Command Line Interface (CLI) tool installed in PATH.
```

What do you think?

**Review by ptgott — APPROVED** [review]

**Review by smallinsky — APPROVED** [review]
